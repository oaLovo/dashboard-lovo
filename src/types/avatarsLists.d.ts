export type ListDataProps = {
  avatar?: string
  name: string
  subtitle: string
}

export interface AvatarsListsProps {
  listData: ListDataProps[]
}
