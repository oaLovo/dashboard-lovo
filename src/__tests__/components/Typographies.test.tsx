import { render } from '@testing-library/react'
import {
  StyledH1,
  StyledH2,
  StyledH3,
  StyledP,
  StyledSpan,
  StyledUl,
} from '@/components'
import { Theme, TypographiesProps } from '@/types'
import { ThemeProvider } from 'styled-components'
import { themesList } from '@/resources/themesList'

const customTypographiesProps: TypographiesProps = {
  color: 'green',
  size: 35,
  lineheight: 40,
  weight: 600,
}

describe('Typographies Component', () => {
  const renderH1Component = (theme: Theme, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledH1 {...props} />
      </ThemeProvider>
    )

  const renderH2Component = (theme: Theme, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledH2 {...props} />
      </ThemeProvider>
    )

  const renderH3Component = (theme: Theme, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledH3 {...props} />
      </ThemeProvider>
    )

  const renderPComponent = (theme: Theme, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledP {...props} />
      </ThemeProvider>
    )

  const renderSpanComponent = (theme: Theme, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledSpan {...props} />
      </ThemeProvider>
    )

  const renderUlComponent = (theme: Theme, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledUl {...props} />
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      // H1
      it('should match the snapshot - H1', () => {
        const { asFragment } = renderH1Component(theme)
        expect(asFragment()).toMatchSnapshot()
      })
      it('should match the snapshot - H1 with custom props', () => {
        const { asFragment } = renderH1Component(theme, customTypographiesProps)
        expect(asFragment()).toMatchSnapshot()
      })

      // H2
      it('should match the snapshot - H2', () => {
        const { asFragment } = renderH2Component(theme)
        expect(asFragment()).toMatchSnapshot()
      })
      it('should match the snapshot - H2 with custom props', () => {
        const { asFragment } = renderH2Component(theme, customTypographiesProps)
        expect(asFragment()).toMatchSnapshot()
      })

      // H3
      it('should match the snapshot - H3', () => {
        const { asFragment } = renderH3Component(theme)
        expect(asFragment()).toMatchSnapshot()
      })
      it('should match the snapshot - H3 with custom props', () => {
        const { asFragment } = renderH3Component(theme, customTypographiesProps)
        expect(asFragment()).toMatchSnapshot()
      })

      // P
      it('should match the snapshot - P', () => {
        const { asFragment } = renderPComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })
      it('should match the snapshot - P with custom props', () => {
        const { asFragment } = renderPComponent(theme, customTypographiesProps)
        expect(asFragment()).toMatchSnapshot()
      })

      // SPAN
      it('should match the snapshot - Span', () => {
        const { asFragment } = renderSpanComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })
      it('should match the snapshot - Span with custom props', () => {
        const { asFragment } = renderSpanComponent(
          theme,
          customTypographiesProps
        )
        expect(asFragment()).toMatchSnapshot()
      })

      // UL
      it('should match the snapshot - UL', () => {
        const { asFragment } = renderUlComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })
      it('should match the snapshot - UL with custom props', () => {
        const { asFragment } = renderUlComponent(theme, customTypographiesProps)
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
