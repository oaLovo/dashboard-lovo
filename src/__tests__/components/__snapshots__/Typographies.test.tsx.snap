// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typographies Component Dark Theme should match the snapshot - H1 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1.5rem;
  font-weight: 600;
  letter-spacing: -0.0625rem;
  line-height: 2.25rem;
}

<h1
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - H1 with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  letter-spacing: -0.0625rem;
  line-height: 2.5rem;
}

<h1
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - H2 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
}

<h2
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - H2 with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

<h2
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - H3 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
}

<h2
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - H3 with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

<h2
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - P 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c0.error {
  color: #FF0202;
}

.c0.success {
  color: #008000;
}

<p
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - P with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

.c0.error {
  color: #FF0202;
}

.c0.success {
  color: #008000;
}

<p
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - Span 1`] = `
<DocumentFragment>
  .c0 {
  color: #89A7CE;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

<span
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - Span with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

<span
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - UL 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
  list-style-position: inside;
}

.c0 li {
  list-style-position: outside;
  margin-left: 0.9375rem;
}

<ul
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Dark Theme should match the snapshot - UL with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
  list-style-position: inside;
}

.c0 li {
  list-style-position: outside;
  margin-left: 0.9375rem;
}

<ul
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - H1 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1.5rem;
  font-weight: 600;
  letter-spacing: -0.0625rem;
  line-height: 2.25rem;
}

<h1
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - H1 with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  letter-spacing: -0.0625rem;
  line-height: 2.5rem;
}

<h1
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - H2 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
}

<h2
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - H2 with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

<h2
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - H3 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
}

<h2
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - H3 with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

<h2
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - P 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

.c0.error {
  color: #FF0202;
}

.c0.success {
  color: #008000;
}

<p
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - P with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

.c0.error {
  color: #FF0202;
}

.c0.success {
  color: #008000;
}

<p
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - Span 1`] = `
<DocumentFragment>
  .c0 {
  color: #828282;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
}

<span
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - Span with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
}

<span
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - UL 1`] = `
<DocumentFragment>
  .c0 {
  color: inherit;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5rem;
  list-style-position: inside;
}

.c0 li {
  list-style-position: outside;
  margin-left: 0.9375rem;
}

<ul
    class="c0"
  />
</DocumentFragment>
`;

exports[`Typographies Component Light Theme should match the snapshot - UL with custom props 1`] = `
<DocumentFragment>
  .c0 {
  color: green;
  font-size: 2.1875rem;
  font-weight: 600;
  line-height: 2.5rem;
  list-style-position: inside;
}

.c0 li {
  list-style-position: outside;
  margin-left: 0.9375rem;
}

<ul
    class="c0"
    color="green"
    lineheight="40"
    size="35"
    weight="600"
  />
</DocumentFragment>
`;
